Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -289,6 +289,12 @@ public struct ApolloCodegenConfiguration: Codable, Equatable {
     ///
     /// See `APQConfig` for more information on Automatic Persisted Queries.
     public let apqs: APQConfig
+    /// Annotate generated Swift code with the Swift `available` attribute and `deprecated`
+    /// argument for parts of the GraphQL schema annotated with the built-in `@deprecated`
+    /// directive.
+    ///
+    /// This is currently limited to enum values.
+    public let warningsOnDeprecatedUsage: Composition
```

Comment:
I initially understood `warningsOnDeprecatedUsage` as an option to display warnings during the codegen itself (not as a way to add annotations in the generated code). Maybe it's worth doing the distinction? Warning during codegen could be used together with something like "warningAsErrors" to fail the CI in case a field becomes deprecated?

Reply:
I couldn't settle on a clean name for the new option but yes it could be better stating it'll result in Swift compiler warnings; `swiftDeprecationWarnings` or similar? I personally don't think there is much value in warning during code generation, those logs are easy to ignore.

Reply:
> swiftDeprecationWarnings or similar?

I can suggest `addAvailableAnnotations`, `annotateDeprecatedUsages` or if a noun is needed `annotationOnDeprecatedUsages`. I wouldn't trust me on this though as I'm not familiar enough with the ecosystem to know what would feel "natural"

 > those logs are easy to ignore

True. It's only useful with something like `allWarningsAsErrors` for catching these early in the CI

Reply:
This is specifically within the `OutputOptions` struct, so it seems clear to me, but fair point that others may not make that assumption. I don't think any of the other name suggestions here are more clear than what we have honestly. The added documentation should be enough here IMO.